front/src/router/index.js

```diff
@@ -10,69 +10,86 @@ import firebase from '@/firebase/init';
 Vue.use(VueRouter);
 
 const routes = [
-  {
-    path: '/',
-    name: 'home',
-    meta: { requireAuth: true },
-    component: Home
-  },
-  {
-    path: '/logout',
-    name: 'logout',
-    meta: { requireAuth: true },
-    component: Logout
-  },
-  {
-    path: '/login',
-    name: 'login',
-    meta: { requireAuth: false },
-    component: Login
-  },
-  {
-    path: '/register',
-    name: 'register',
-    meta: { requireAuth: false },
-    component: Register
-  },
-  {
-    path: '/about',
-    name: 'about',
-    meta: { requireAuth: false },
-    component: About
-  },
-  {
-    path: '/teams',
-    name: 'teams',
-    meta: { requireAuth: true },
-    component: () =>
-      import(/* webpackChunkName: "about" */ '../views/Teams.vue')
-  },
-  {
-    path: '/gifs',
-    name: 'gifs',
-    meta: { requireAuth: true },
-    component: () => import(/* webpackChunkName: "about" */ '../views/Gifs.vue')
-  },
-  {
-    path: '/schedule',
-    name: 'schedule',
-    meta: { requireAuth: true },
-    component: () =>
-      import(/* webpackChunkName: "about" */ '../views/Schedule.vue')
-  }
+    {
+        path: '/',
+        name: 'home',
+        meta: { requireAuth: true },
+        component: Home,
+    },
+    {
+        path: '/logout',
+        name: 'logout',
+        meta: { requireAuth: true },
+        component: Logout,
+    },
+    {
+        path: '/login',
+        name: 'login',
+        meta: { requireAuth: false },
+        component: Login,
+    },
+    {
+        path: '/register',
+        name: 'register',
+        meta: { requireAuth: false },
+        component: Register,
+    },
+    {
+        path: '/about',
+        name: 'about',
+        meta: { requireAuth: false },
+        component: About,
+    },
+    {
+        path: '/teams',
+        name: 'teams',
+        meta: { requireAuth: true },
+        component: () =>
+            import(/* webpackChunkName: "about" */ '../views/Teams.vue'),
+    },
+    {
+        path: '/gifs',
+        name: 'gifs',
+        meta: { requireAuth: true },
+        component: () =>
+            import(/* webpackChunkName: "about" */ '../views/Gifs.vue'),
+    },
+    {
+        path: '/schedule',
+        name: 'schedule',
+        meta: { requireAuth: true },
+        component: () =>
+            import(/* webpackChunkName: "about" */ '../views/Schedule.vue'),
+    },
 ];
 
 const router = new VueRouter({
-  routes
+    routes,
 });
 
+let isFirstTransition = true;
+
 router.beforeEach(async (to, from, next) => {
-  const requireAuth = to.matched.some(record => record.meta.requireAuth);
-  if (requireAuth && !(await firebase.getCurrentUser())) {
-    next('login');
-  } else {
-    next();
-  }
+    const lastRouteName = localStorage.getItem('lastToPath');
+
+    const shouldRedirect = Boolean(
+        to.name === 'home' && lastRouteName && isFirstTransition
+    );
+
+    if (shouldRedirect) next({ name: lastRouteName });
```

Comment:
i prefer to have brackets even for one line :)

Reply:
does next prevent the end of the method execution or do we have to force exit with a `return` ?

Reply:
Next force the end of the method. 🙂